src/sha1/sha1.c

```diff
@@ -0,0 +1,360 @@
+ *      sha Error Code.
+ *
+ */
+int SHA1Result(SHA1Context *context,
+               uint8_t Message_Digest[SHA1HashSize])
+{
+	int i;
+
+	if (!context || !Message_Digest) {
+		return shaNull;
+	}
+
+	if (context->Corrupted) {
+		return context->Corrupted;
+	}
+
+	if (!context->Computed) {
+		SHA1PadMessage(context);
+		for (i = 0; i < 64; ++i) {
+			/* message may be sensitive, clear it out */
+			context->Message_Block[i] = 0;
+		}
+		context->Length_Low = 0; /* and clear length */
+		context->Length_High = 0;
+		context->Computed = 1;
+	}
+
+	for (i = 0; i < SHA1HashSize; ++i) {
+		Message_Digest[i] = context->Intermediate_Hash[i >> 2] >> 8 * (3 - (i & 0x03));
+	}
+
+	return shaSuccess;
+}
+
+/*
+ *  SHA1Input
+ *
+ *  Description:
+ *      This function accepts an array of octets as the next portion
+ *      of the message.
+ *
+ *  Parameters:
+ *      context: [in/out]
+ *          The SHA context to update
+ *      message_array: [in]
+ *          An array of characters representing the next portion of
+ *          the message.
+ *      length: [in]
+ *          The length of the message in message_array
+ *
+ *  Returns:
+ *      sha Error Code.
+ *
+ */
+int SHA1Input(SHA1Context *context,
+              const uint8_t *message_array,
+              unsigned length)
+{
+	if (!length) {
+		return shaSuccess;
+	}
+
+	if (!context || !message_array) {
+		return shaNull;
+	}
+
+	if (context->Computed) {
+		context->Corrupted = shaStateError;
+
+		return shaStateError;
+	}
+
+	if (context->Corrupted) {
+		return context->Corrupted;
+	}
+	while (length-- && !context->Corrupted) {
+		context->Message_Block[context->Message_Block_Index++] =
+		    (*message_array & 0xFF);
+
+		context->Length_Low += 8;
+		if (context->Length_Low == 0) {
+			context->Length_High++;
+			if (context->Length_High == 0) {
+				/* Message is too long */
+				context->Corrupted = 1;
+			}
+		}
+
+		if (context->Message_Block_Index == 64) {
+			SHA1ProcessMessageBlock(context);
+		}
+
+		message_array++;
+	}
+
+	return shaSuccess;
+}
+
+/*
+ *  SHA1ProcessMessageBlock
+ *
+ *  Description:
+ *      This function will process the next 512 bits of the message
+ *      stored in the Message_Block array.
+ *
+ *  Parameters:
+ *      None.
+ *
+ *  Returns:
+ *      Nothing.
+ *
+ *  Comments:
+
+ *      Many of the variable names in this code, especially the
+ *      single character names, were used because those were the
+ *      names used in the publication.
+ *
+ *
+ */
+void SHA1ProcessMessageBlock(SHA1Context *context)
+{
+	const uint32_t K[] = {/* Constants defined in SHA-1   */
+	                      0x5A827999,
+	                      0x6ED9EBA1,
+	                      0x8F1BBCDC,
+	                      0xCA62C1D6};
+	int t;                  /* Loop counter                */
+	uint32_t temp;          /* Temporary word value        */
+	uint32_t W[80];         /* Word sequence               */
+	uint32_t A, B, C, D, E; /* Word buffers                */
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define each identifier in a dedicated statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3ASingleDeclarationPerStatement)